checkstyle-tester/launch.groovy

```diff
@@ -0,0 +1,191 @@
+        project ->
+            if (!project.startsWith('#') && !project.isEmpty()) {
+                def params = project.split('\\|')
+                def repoName = params[REPO_NAME_PARAM_NO]
+                def repoType = params[REPO_TYPE_PARAM_NO]
+                def repoUrl = params[REPO_URL_PARAM_NO]
+                def commitId = params[REPO_COMMIT_ID_PARAM_NO]
+                if (params.length == FULL_PARAM_LIST_SIZE) {
+                    excludes += ',' + params[REPO_EXCLUDES_PARAM_NO]
+                }
+                cloneRepository(repoName, repoType, repoUrl, commitId, srcDir)
+            }
+    }
+    runMavenExecution(srcDir, excludes, checkstyleConfig)
+    postProcessCheckstyleReport()
+}
+
+def cloneRepository(repoName, repoType, repoUrl, commitId, srcDir) {
+    def srcDestinationDir = "$srcDir/$repoName"
+    if (!Files.exists(Paths.get(srcDestinationDir))) {
+        def cloneCmd = getCloneCmd(repoType, repoUrl, srcDestinationDir)
+        println "Cloning $repoType repository '$repoName' to $srcDestinationDir folder ..."
+        executeCmd(cloneCmd)
+        println "Cloning $repoType repository '$repoName' - completed\n"
+    }
+    if (commitId && commitId != '') {
+        def resetCmd = getResetCmd(repoType, commitId)
+        println "Reseting $repoType sources to commit '$commitId'"
+        executeCmd(resetCmd, new File("$srcDestinationDir"))
+    }
+    println "$repoName is synchronized\n"
+}
+
+def getCloneCmd(repoType, repoUrl, srcDestinationDir) {
+    def cloneCmd = ''
+    switch (repoType) {
+        case 'git':
+            cloneCmd = "git clone $repoUrl $srcDestinationDir"
+            break
+        case 'hg':
+            cloneCmd = "hg clone $repoUrl $srcDestinationDir"
+            break
+        default:
+            println "Error! Unknown $repoType repository."
+            break
+    }
+}
+
+def getResetCmd(repoType, commitId) {
+    def resetCmd = ''
+    switch (repoType) {
+        case 'git':
+            resetCmd = "git reset --hard $commitId"
+            break
+        case 'hg':
+            resetCmd = "hg up $commitId"
+            break
+        default:
+            println "Error! Unknown $repoType repository."
+            break
+    }
+}
+
+def runMavenExecution(srcDir, excludes, checkstyleConfig) {
+    println "Running 'mvn clean' on $srcDir ..."
+    def mvnClean = "mvn --batch-mode clean"
+    executeCmd(mvnClean)
+    println "Running Checkstyle on $srcDir ... with excludes $excludes"
+    def mvnSite = "mvn -e --batch-mode site -Dcheckstyle.config.location=$checkstyleConfig -Dcheckstyle.excludes=$excludes -DMAVEN_OPTS=-Xmx3024m"
+    executeCmd(mvnSite)
+    println "Running Checkstyle on $srcDir - finished"
+}
+
+def postProcessCheckstyleReport() {
+    println 'linking report to index.html'
+    new File('target/site/index.html').renameTo 'target/site/_index.html'
+    Files.createLink(Paths.get("target/site/index.html"), Paths.get("target/site/checkstyle.html"))
```

Comment:
It should work. http://stackoverflow.com/questions/12509132/how-to-force-java-7-to-create-directory-symlinks-on-windows-instead-of-file-s?rq=1
I've played with another utility that creates symlinks on Windows 7, so it is supported, just not easily accessible.
http://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/

rename in windows is: `ren`. mv in windows is: `move`.